### PR TITLE
Unit tests TestNG

### PR DESCRIPTION
**Pull request by VictoriaBadenko**
1. Added unit tests for JsonParser class.
2. Added 1 test for RealItem class.
3. Added 1 test for VirtualItem class.
4. Added 2 tests for Cart class.
5. Disabled one of tests from point 1.
6. Added 4 testng.xml file with including groups/excluding groups/including classes/excluding classes/with parameters/etc.
7. Added grouped assertion for one of tests.

### CODE REVIEW

File: testNgExcludeClassAndParameters.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <test name="Test with Excluding Classes and Parameters">
+        <classes>
+            <class name="RealItemTest"/>
+            <class name="VirtualItemTest"/>
+            <class name="JsonParserTests"/>
+            <methods>
+                <exclude name=".*"/>
+            </methods>
+        </classes>
+        <parameter name="environment" value="test"/>
```
Comment (LikBez123) [inline testNgExcludeClassAndParameters.xml]: I did not find the use of this parameter in the tests. Could you add it somewhere?

**Review by LikBez123 — CHANGES_REQUESTED** [review]

File: testNgExcludeClassAndParameters.xml (hunk 1 above)
Comment (VictoriaBadenko) [reply testNgExcludeClassAndParameters.xml]: Added parameters to class RealItemTest.

File: testNgExcludeClassAndParameters.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<suite name="Excluding Classes and Parameters Suite">
+    <test name="Test with Excluding Classes and Parameters">
+        <classes>
+            <class name="RealItemTest"/>
+            <class name="VirtualItemTest"/>
+            <class name="JsonParserTests"/>
+            <methods>
+                <exclude name=".*"/>
```
Comment (LikBez123) [inline testNgExcludeClassAndParameters.xml]: try to exclude not all methods, but only one by name

**Review by LikBez123 — APPROVED** [review]

File: testNgExcludeClassAndParameters.xml (hunk 2 above)
Comment (VictoriaBadenko) [reply testNgExcludeClassAndParameters.xml]: Done

File: testNgIncludeGroup.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd">
+<suite name="Including Groups Suite">
```
Comment (victortylikov) [inline testNgIncludeGroup.xml:2]: @VictoriaBadenko please run this suite and check test results, what's wrong with it?
Screenshot 2023-08-03 at 1 49 49 PM

**Review by victortylikov — CHANGES_REQUESTED** [review]

File: testNgIncludeGroup.xml (hunk 3 above)
Comment (VictoriaBadenko) [reply testNgIncludeGroup.xml:2]: Fixed this issue.

**Review by victortylikov — APPROVED** [review]

**Pull request merged**